src/Compilers/CSharp/Test/Semantic/SourceGeneration/GeneratorDriverTests.cs

```diff
@@ -2907,5 +2907,109 @@ public void Incremental_Generators_Can_Recover_From_Exceptions()
+
+            var timing = driver.GetTimingInfo();
+
+            Assert.Equal(TimeSpan.Zero, timing.ElapsedTime);
+
+            var generatorTiming = Assert.Single(timing.GeneratorTimes);
+            Assert.Equal(generator, generatorTiming.Generator);
+            Assert.Equal(TimeSpan.Zero, generatorTiming.ElapsedTime);
+        }
+
+        [Fact]
+        public void Can_Get_Timing_Info()
+        {
+            var source = "class C{}";
+
+            var parseOptions = TestOptions.RegularPreview;
+            Compilation compilation = CreateCompilation(source, options: TestOptions.DebugDll, parseOptions: parseOptions);
+            compilation.VerifyDiagnostics();
+
+            var generator = new PipelineCallbackGenerator(ctx =>
+            {
+                ctx.RegisterSourceOutput(ctx.CompilationProvider, (context, text) =>
+                {
+                    context.AddSource("generated", "");
+                });
+            }).AsSourceGenerator();
+
+            GeneratorDriver driver = CSharpGeneratorDriver.Create(new[] { generator }, parseOptions: parseOptions, driverOptions: new GeneratorDriverOptions(IncrementalGeneratorOutputKind.None, trackIncrementalGeneratorSteps: true));
+
+            driver = driver.RunGenerators(compilation);
+            var timing = driver.GetTimingInfo();
+
+            Assert.NotEqual(TimeSpan.Zero, timing.ElapsedTime);
+
+            var generatorTiming = Assert.Single(timing.GeneratorTimes);
+            Assert.Equal(generator, generatorTiming.Generator);
+            Assert.NotEqual(TimeSpan.Zero, generatorTiming.ElapsedTime);
+            Assert.True(timing.ElapsedTime >= generatorTiming.ElapsedTime);
+        }
+
+        [Fact]
+        public void Can_Get_Timing_Info_From_Multiple_Generators()
+        {
+            var source = "class C{}";
+
+            var parseOptions = TestOptions.RegularPreview;
+            Compilation compilation = CreateCompilation(source, options: TestOptions.DebugDll, parseOptions: parseOptions);
+            compilation.VerifyDiagnostics();
+
+            var generator = new PipelineCallbackGenerator(ctx =>
+            {
+                ctx.RegisterSourceOutput(ctx.CompilationProvider, (context, text) =>
+                {
+                    context.AddSource("generated", "");
+                });
+            }).AsSourceGenerator();
+
+            var generator2 = new PipelineCallbackGenerator2(ctx =>
+            {
+                ctx.RegisterSourceOutput(ctx.CompilationProvider, (context, text) =>
+                {
+                    context.AddSource("generated", "");
+                });
+            }).AsSourceGenerator();
+
+            GeneratorDriver driver = CSharpGeneratorDriver.Create(new[] { generator, generator2 }, parseOptions: parseOptions, driverOptions: new GeneratorDriverOptions(IncrementalGeneratorOutputKind.None, trackIncrementalGeneratorSteps: true));
+
+            driver = driver.RunGenerators(compilation);
```

Comment:
Should we have a test where you run twice and we confirm it's only counting the most recent run?

Reply:
Good idea.